service/src/main/java/bio/terra/cbas/dependencies/wes/CromwellService.java

```diff
@@ -128,6 +127,35 @@ public static String getErrorMessage(List<FailureMessage> failureMessages) {
     }
   }
 
+  /**
+   * Cromwell accepts an object "Workflow Options" to specify additional configuration for a
+   * workflow. Here, we build that object with the parameters we care about. final_workflow_log_dir
+   * specifies the path where outputs will be written. write_to_cache and read_from_cache are both
+   * related to call caching. When a user enables call caching, these should both be set to true.
+   * Otherwise, they should both be set to false.
+   * https://cromwell.readthedocs.io/en/stable/wf_options/Overview/ for more info.
+   *
+   * @param isCallCachingEnabled Whether the user wishes to run this workflow with call caching.
+   * @return A string formatted as a JSON object that can be used as cromwell's Workflow Options.
+   * @throws JsonProcessingException
+   */
+  public static String buildWorkflowOptionsJson(
+      Optional<String> finalWorkflowLogDir, Boolean isCallCachingEnabled)
+      throws JsonProcessingException {
+    if (isCallCachingEnabled == null) {
+      log.warn("Null call caching parameter provided. Defaulting to false.");
+      isCallCachingEnabled = false;
+    }
```

Comment:
Maybe this check can be moved to RunSetsApiController since the field is associated with RunSet and additionally this doesn't get checked for all the workflows in the RunSet?

Reply:
Additionally maybe `buildWorkflowOptionsJson` can also be refactored to run once per RunSet if the workflow options are going to be same for all workflows in the RunSet?

Reply:
Makes sense to me, happy to make that change.